sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/readme.md

```diff
@@ -0,0 +1,149 @@
+- Integration with the Digital Twins Model Parser
+
+## Initializing the models repository client
+
+### Simple Initialization
+
+```C# Snippet:IoTModelsRepositorySamplesCreateServiceClientSimpleWithGlobalEndpoint
+// When no Uri is provided for instantiation, the Azure IoT Models Repository global endpoint
+// https://devicemodels.azure.com/ is used and the dependency model resolution option is set to TryFromExpanded.
+var client = new ModelsRepositoryClient(new ModelsRepositoryClientOptions());
+Console.WriteLine($"Initialized client pointing to global endpoint: {client.RepositoryUri}");
+```
+
+```C# Snippet:IoTModelsRepositorySamplesCreateServiceClientSimpleWithLocalRepository
+// The client will also work with a local filesystem Uri. This example shows initalization
+// with a local Uri and disabling model dependency resolution.
+client = new ModelsRepositoryClient(new Uri(ClientSamplesLocalModelsRepository),
+    new ModelsRepositoryClientOptions(resolutionOption: DependencyResolutionOption.Disabled));
+Console.WriteLine($"Initialized client pointing to local path: {client.RepositoryUri}");
+```
+
+### Override options
+
+If you need to override pipeline behavior, such as provide your own `HttpClient` instance, you can do that via constructor that takes a [ModelsRepositoryClientOptions][modelsrepository_clientoptions] parameter.
+It provides an opportunity to override default behavior including:
+
+- Overriding [transport][azure_core_transport]
+- Enabling [diagnostics][azure_core_diagnostics]
+- Controlling [retry strategy](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Configuration.md)
+
+## Publish Models
+
+Publishing models to the models repository requires [exercising][modelsrepository_publish_msdocs] common GitHub workflows.
+
+## Get Models
+
+After publishing, your model(s) will be available for consumption from the global repository endpoint. The following snippet shows how to retrieve the corresponding JSON-LD content.
+
+```C# Snippet:IoTModelsRepositorySamplesGetModelsFromGlobalRepoAsync
+var dtmi = "dtmi:com:example:TemperatureController;1";
+
+// Global endpoint client
+var client = new ModelsRepositoryClient();
+
+// The output of GetModelsAsync() will include at least the definition for the target dtmi.
+// If the dependency model resolution option is not disabled, then models in which the
+// target dtmi depends on will also be included in the returned IDictionary<string, string>.
+IDictionary<string, string> models = await client.GetModelsAsync(dtmi).ConfigureAwait(false);
+
+// In this case the above dtmi has 2 model dependencies.
+// dtmi:com:example:Thermostat;1 and dtmi:azure:DeviceManagement:DeviceInformation;1
+Console.WriteLine($"{dtmi} resolved in {models.Count} interfaces.");
+```
+
+Because GitHub pull-request workflows are a core aspect of the IoT Models Repository service, and to supported related use cases, the client supports getting models from a local repository.
```

Comment:
I don't understand this sentence. It talks about GH and PRs, but then pivots to local repo. What do we mean by that? A custom repo on GH?

A local repo would be Git, not GitHub. That seems the same as just using a local file system, though.

Can you clarify?

Reply:
It is referring to the typical PR workflow when contributing to a repository on GitHub. You fork the target repo via GitHub, clone your fork locally using Git and work with/manipulate files or content locally from your clone. In the context of the model repository, cloning means cloning the repo locally...hence local repository. Therefore the client supports model resolution locally so you can build models and validate them against the clone.

Reply:
Oh, you mean test against local changes of a cloned repo?

I take local repo to mean something different.

Reply:
If you can clarify the comment, I think it'd help.

Reply:
Will do

Reply:
I re-wrote this description to explain more. Do you think it is clear now?